### PR TITLE
[stable/prometheus-operator] script for syncing prometheus rules from coreos and etcd upstream

### LINKED ISSUE #8749: [stable/prometheus-operator] Alert rules are created for components even if scrapes are disabled
*Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature Request

**Version of Helm and Kubernetes**:
*Helm*: 2.10.0
*Kubernetes*: 1.10.6

**Which chart**:
stable/prometheus

**What happened**:
If scraping of a particular Kubernetes components (e.g. API server) is disabled via Helm values, corresponding alert rules are still configured in Prometheus leading to alerts that are constantly firing.

**What you expected to happen**:
If scraping configurations are disabled then alert rules that would rely on those metrics should also be disabled.

**How to reproduce it** (as minimally and precisely as possible):
- Deploy the prometheus operator chart with the below values set:
  ```
  kubeControllerManager:
    enabled: false
  ```
- The below alert rule is still configured in the  `PrometheusRule` resource and will immediately enter an alerting state:
  ```
    - alert: KubeControllerManagerDown
      annotations:
        message: KubeControllerManager has disappeared from Prometheus target discovery.
        runbook_url: https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubecontrollermanagerdown
      expr: |
        absent(up{job="kube-controller-manager"} == 1)
      for: 15m
      labels:
        severity: critical
  ```

### PR DESCRIPTION
**Pull request by paskal**
#### What this PR does / why we need it:
This PR:
- add python scripts for alertmanager rules and Grafana dashboards sync ([this one](https://github.com/coreos/prometheus-operator/blob/master/helm/hack/sync_kube_prometheus.py) used as starting point, idea from [this comment](https://github.com/helm/charts/issues/8749#issuecomment-436389693) by @gianrubio)
- sync existing rules and dashboards from etcd and coreos
- append turn on\off conditions on sync on proper alerts and dashboards files, and on particular alerts alerts
- add hack folder readme
- add readme section about where Grafana dashboard and Prometheur rules come from

#### Which issue this PR fixes
  - fixes #8749

#### Special notes for your reviewer:
/cc @vsliouniaev @brancz for review.
Added Python script is working with current version of the files and gives ability to reliably sync up rules with upstream of coreos and etcd. It may break in the future with changes on coreos side, but is much better than nothing.
I've tested the changes and fixed all found bugs, expected behavior after upgrade: a few new alerts appear (because of the sync) and a few are removed (due to enabled conditional enable like `{{- if and .Values.defaultRules.create .Values.nodeExporter.enabled }}`).
Also reviewer might run `stable/prometheus-operator/hacks/*.py` himself, expected result: either nothing changes, or new changes from coreos repo will be introduced to files.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/prometheus-operator/hack/sync_prometheus_rules.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        scalar.style = style
+        return scalar
+
+    return new_representer
+
+
+# Source files list
+charts = [
+    {
+        'source': 'https://raw.githubusercontent.com/coreos/prometheus-operator/master/contrib/kube-prometheus/manifests/prometheus-rules.yaml',
```
Comment (brancz) [inline stable/prometheus-operator/hack/sync_prometheus_rules.py:28]: Note that eventually we will stop checking in the generated code and an artifact will only be available on kube-prometheus releases. That doesn't prevent this strategy, just means updating is likely going to be possible less frequently.

**Review by brancz — COMMENTED** [review]
I don't use or maintain helm charts so I can't comment on the result, but the general strategy sounds and looks promising.

File: stable/prometheus-operator/hack/sync_prometheus_rules.py (hunk 1 above)
Comment (paskal) [reply stable/prometheus-operator/hack/sync_prometheus_rules.py:28]: Thanks for review! Sounds good enough for me. Maybe we should to add a note about running rules sync script on appVersion bump, and maybe add check to CI run - I'll see what I can do.

**Comment by paskal** [comment]
@gianrubio, @vsliouniaev refused to review because he couldn't review Python code. Seems it's all up to you, could you please look into this PR and tell me if there is anything I should change before merging?

**Comment by paskal** [comment]
@gianrubio, please review and merge. I've added dashboards sync, so now this repo Grafana dashboards and alertmanager rules are all imported and should not be modified here, but rather in original locations and then imported back.
I've tested all these changes as hard as I could, it works on my production. Only thing I couldn't test is etcd Grafana dashboard, however I think it's better it to be imported from [known location](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/grafana.json), and fixed there than have some outdated version that works but couldn't be changed.
If you have any questions, please reach me - I've put a lot of effort in this PR and I want the community to benefit from the work I've done.

**Comment by paskal** [comment]
@gianrubio, could you please leave any comment on this one, what are your plans? Will you test it and then merge, or are you waiting for something to happen, and can I assist in that?

**Comment by paskal** [comment]
@davidkarlsen could you look into it please?

**Comment by paskal** [comment]
@davidkarlsen @gianrubio I can separate this PR into two, first one will just add scripts and adjust readme, and other one will update rules using these scripts (after first one will be merged). This one does both.

**Comment by paskal** [comment]
Closing in favor of #10488. This one will be re-created after scripts will be merged into upstream.

**Pull request closed without merging**